src/jdk.management/share/classes/com/sun/management/internal/HotSpotDiagnostic.java

```diff
@@ -167,27 +153,8 @@ public void dumpThreads(String outputFile, ThreadDumpFormat format) throws IOExc
         if (!file.isAbsolute())
             throw new IllegalArgumentException("'outputFile' not absolute path");
 
-        // need ManagementPermission("control")
-        @SuppressWarnings("removal")
-        SecurityManager sm = System.getSecurityManager();
-        if (sm != null)
-            Util.checkControlAccess();
-
         try (OutputStream out = Files.newOutputStream(file, StandardOpenOption.CREATE_NEW)) {
-            PrivilegedExceptionAction<Void> pa = () -> {
                 dumpThreads(out, format);
-                return null;
-            };
-            try {
-                AccessController.doPrivileged(pa);
-            } catch (PrivilegedActionException pae) {
-                Throwable cause = pae.getCause();
-                if (cause instanceof IOException ioe)
-                    throw ioe;
-                if (cause instanceof RuntimeException e)
-                    throw e;
-                throw new RuntimeException(cause);
```

Comment:
I think you've made an assumption that these are all related to lack of prividleges. I don't think that is the case, and they are all exceptions that can result from calling dumpThreads(). I think you need a try block around the dumpThreads() call that handles all these exceptions in the same manner.

Reply:
OK yes it looks that way but I think we're OK:

Previously, IOException and RuntimeException were rethrown, and that is I think everything we expected to see wrapped in an PrivilegedActionException.
Anything else was wrapped in a RuntimeException, but outside of IOException which was declared to be thrown, surely it's going to be an unchecked RuntimeException...


Here in dumpThreads we declare IOException and Runtime is unchecked.
So those two get thrown as before, not sure if we need to catch anything else.


Experimenting: invoking dumpHeap through the HotSpotDiagnostic MBean, I can see:

javax.management.RuntimeMBeanException: java.lang.IllegalArgumentException: heapdump file must have .hprof extension
or
javax.management.MBeanException: java.io.IOException: Permission denied


It looks like IO or IAException (which is a RuntimeException) get handled OK, and the MBean invocation, which called dumpHeap, wraps them in other exceptions.